tsdb/docs/format/wal.md

```diff
@@ -50,6 +50,27 @@ Series records encode the labels that identifies a series and its unique ID.
 └────────────────────────────────────────────┘
 ```
 
+### Metadata records
+
+Metadata records encode the metadata associated with a series.
+
+```
+┌────────────────────────────────────────────┐
+│ type = 1 <1b>                              │
+├────────────────────────────────────────────┤
+│ ┌────────────────────────────────────────┐ │
+│ │ id <uvarint>                           │ │
+│ ├─────────────────────┬──────────────────┤ │
+│ │ len(type) <uvarint> │ type <bytes>     │ │
+│ ├─────────────────────┴──────────────────┤ │
+│ │ len(unit) <uvarint> │ unit <bytes>     │ │
+│ ├─────────────────────┴──────────────────┤ │
+│ │ len(help) <uvarint> │ help <bytes>     │ │
```

Comment:
Should we allow for additional metadata in future?

Reply:
That sounds good, I can add a versioning scheme to record types if you like. It may add some complexity and doesn't exist already - but if that's the direction that would be ideal to move towards I can attempt implementing a relatively simple method to try to achieve this. 

Otherwise we'd have to change the record type altogether (i.e. MetadataV2) which means on a rollback of version, all new MetadataV2 records would be skipped (which would not be ideal, ideally you want to partially read the fields you know about, just skip the new fields you don't know about).

Reply:
Don't add strong versioning, that just means keeping multiple parsers around. I'd add the number of entries that follow.

Reply:
After catching up, went with writing the size of the encoded fields upfront so we can skip any fields unknown to the decoder once we've decoded all known fields to the decoder.

I've added a unit test to cover this case.